src/stores/globalStore.js

```diff
@@ -51,7 +52,61 @@ export const globalStore = createStore(
   {
     logout(state) {
       userStorage.reset();
-      return { ...state, currentUser: null, loggedIn: false };
+      return { ...state, currentUser: null };
     },
+    createPost(state, content) {
+      if (!state.currentUser) {
+        return { ...state };
+      }
+
+      const postIds = state.posts.map((post) => post.id);
+      const post = {
+        id: generateNewId(postIds),
+        author: state.currentUser.username,
+        time: Date.now(),
+        content,
+        likeUsers: [],
+      };
+
+      return {
+        ...state,
+        posts: [post, ...state.posts],
+      };
+    },
+    toggleLikePost(state, postId) {
+      if (!state.currentUser) {
+        return { ...state };
+      }
+
+      const posts = state.posts.map((post) => {
+        if (post.id !== postId) return post;
+
+        const username = state.currentUser.username;
+        const hasItem = post.likeUsers.includes(state.currentUser.username);
+
+        return {
+          ...post,
+          likeUsers: hasItem
+            ? removeItem(post.likeUsers, username)
+            : addItem(post.likeUsers, username),
+        };
+      });
+
+      return { ...state, posts };
+    },
+  },
+  {
+    hasLikedPost: (state, postId) => {
+      if (!state.currentUser) {
+        return false;
+      }
+
+      return (
+        state.posts
+          .find((post) => post.id === postId)
+          ?.likeUsers.includes(state.currentUser.username) ?? false
+      );
+    },
+    loggedIn: (state) => Boolean(state.currentUser),
   },
```

Comment:
### 파생 상태 처리 관련
좋아요를 누른 Post 인지 여부와, 현재 로그인된 상태인지에 대한 정보는 기존에 상태를 가공하여 얻을 수 있는 파생 상태인 것 같아서, 이번 과제 요구사항은 아니지만 selector 를 추가했습니다. (selector 구현 코드 링크)

다만, selector 상태 조회를 위해서는 selector 함수를 매번 호출해주어야 하는데, 어떤 상태가 바뀌면 파생상태가 자동으로 재계산되어 저장되고, selector 함수 재호출 없이도 변경사항으로 인해 리렌더링이 발생하면 좋을 것 같은데, 해당 방향까지는 깊이 고민하지 못해 아쉽습니다.

Reply:
파생 상태 처리에 대해서 selector 패턴을 도입한 점이 너무 인상적이었습니다 굿굿 자동 재계산에 대한 고민도 최고최고bbbb
아니면 간단한 캐싱 메커니즘을 추가하면 어떨까 싶어요?! 변경된 상태 키에 따라 관련 캐시만 초기화하는 방법도 있겠지만 .. 

```suggestion
const selectors = Object.fromEntries(
    Object.entries(initialSelectors).map(([key, selectorFn]) => [
      key,
      (...args) => {
        const cacheKey = `${key}:${JSON.stringify(args)}`;
        
        if (!selectorCache.has(cacheKey)) {
          selectorCache.set(cacheKey, selectorFn(state, ...args));
        }
        
        return selectorCache.get(cacheKey);
      },
    ]),
  );
```

gpt 에게 이런 문제를 겪어서 해결방식으로 캐싱 매커니즘을 추가하고 싶어~ 요런식으로 작성해서 만들어줬는데 예시가 적절할까 모르겠습니다 허허